nexus/src/lib.rs

```diff
@@ -167,6 +167,14 @@ pub async fn run_server(config: &Config) -> Result<(), String> {
         .log
         .to_logger("nexus")
         .map_err(|message| format!("initializing logger: {}", message))?;
+    if let Err(e) = usdt::register_probes() {
+        if config.dtrace.required {
+            return Err(e.to_string());
+        }
+        warn!(log, "failed to register DTrace probes: {}", e);
```

Comment:
Nit: Log before returning the error?

Reply:
I assumed the caller would log or print the error in this case, but that's a good idea.